frappe/database/database.py

```diff
@@ -502,8 +502,15 @@ def get_values_from_single(self, fields, filters, doctype, as_dict=False, debug=
 			r = self.sql("""select field, value
 				from `tabSingles` where field in (%s) and doctype=%s"""
 					% (', '.join(['%s'] * len(fields)), '%s'),
-					tuple(fields) + (doctype,), as_dict=False, debug=debug)
-
+					tuple(fields) + (doctype,), as_dict=False, debug=debug, as_list=True)
+			
+			for field in r:
+				fieldtype = self.sql("""
+					select fieldtype from `tabDocfield` 
```

Comment:
There are a couple of things wrong here:
1. SQL query in a for loop
1. Querying from tabDocField, which doesn't consider fieldtype change via Property Setter.

Are you sure this can't be done using `meta`?

Reply:
1. I can remove the loop and use "and" in the where clause and get all the values together and then process
2. Yes, I tried using meta but I am getting recursion depth errors. The Meta class constructor uses Document class constructor and somewhere there get_value() is used.
3. Since this code only runs for single doctypes it shouldn't be a problem since they aren't customisable as far as I know